### DISCUSSION TITLE
Getting started for next.js contribution

### DISCUSSION THREAD

**Discussion by coder-abhi27** [vercel/next.js#57556] [Help] [answered]
### Summary

I want to contriute to next.js but i am not able to figure out how. I have good knowledge of javascript and typescript. I want to solve bugs and add features. I want is there any documentation that you can help for started. How to setup locally and start debugging. Any example to solve bugs for next.js

**Comment by icyJoseph** [comment]
Everything you need to know is in these documents: https://github.com/vercel/next.js/tree/canary/contributing/core

Typically, when people finally manage to produce a reproduction repository, I use that as base and follow this instructions:

- https://github.com/vercel/next.js/blob/canary/contributing/core/developing-using-local-app.md

To see how changes to the next.js codebase affect the reproduction repository.

**Reply by coder-abhi27** [reply, marked as answer]
Thanks a lot @icyJoseph! let me explore the docs and try to build these
